go/vt/vtgate/planbuilder/testdata/select_cases.json

```diff
@@ -6183,5 +6183,27 @@
         "user.user"
       ]
     }
+  },
+  {
+    "comment": "derived table column selection should only return requested columns",
+    "query": "select t.id from (select id, name from user) as t",
+    "plan": {
+      "Type": "Scatter",
+      "QueryType": "SELECT",
+      "Original": "select t.id from (select id, name from user) as t",
+      "Instructions": {
+        "OperatorType": "Route",
+        "Variant": "Scatter",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "FieldQuery": "select t.id from (select id, `name` from `user` where 1 != 1) as t where 1 != 1",
+        "Query": "select t.id from (select id, `name` from `user`) as t"
+      },
+      "TablesUsed": [
+        "user.user"
+      ]
+    }
```

Comment:
This test also passes already on `main`, so I don't think it exercises the path here that has the bug. Please make sure that we add a test here that actually exercises the problematic case and shows a failing on `main` and then passes on this branch.